src/daemon/CMakeLists.txt

```diff
@@ -37,30 +37,7 @@ set(daemon_sources
 
 set(daemon_headers)
 
-set(daemon_private_headers
-  command_parser_executor.h
-  command_server.h
-  core.h
-  daemon.h
-  executor.h
-  p2p.h
-  protocol.h
-  rpc.h
-  rpc_command_executor.h
-
-  # cryptonote_protocol
-  ../cryptonote_protocol/cryptonote_protocol_defs.h
-  ../cryptonote_protocol/cryptonote_protocol_handler.h
-  ../cryptonote_protocol/cryptonote_protocol_handler.inl
-  ../cryptonote_protocol/cryptonote_protocol_handler_common.h
-
-  # p2p
-  ../p2p/net_node.h
-  ../p2p/net_node_common.h
-  ../p2p/net_peerlist.h
-  ../p2p/net_peerlist_boost_serialization.h
-  ../p2p/p2p_protocol_defs.h
-  ../p2p/stdafx.h)
+monero_find_all_headers(daemon_private_headers "${CMAKE_CURRENT_SOURCE_DIR}")
```

Comment:
Is it okay to use monero_find_all_headers here since it won't capture the ../cryptonote_protocol and ../p2p headers?

Reply:
They are (and should be) collected by their individual CMakeLists.txt and only exported to the dependent libraries. In our case, the headers are being exported because the `src` directory is the "header directory". Therefore we access these headers via:
`#include "p2p/p2p_protocol_defs.h"`
and not just:
`#include "p2p_protocol_defs.h"`

Reply:
And if only the `GNU make` discovers that any of the headers changes, it will recompile the source code, whether they are explicitly mentioned as before or not. 

The difference is in the IDE itself. If a header is not captured, it won't be visible in the IDE anymore, and even if you open the header in the IDE, change it and expect the dependent sources to compile, they won't since the IDE wouldn't autosave the header's changes, thus giving no signal to `GNU make`. I checked both the `p2p` and `cryptonote_protocol` and their files are all globed.

Reply:
> They are (and should be) collected by their individual CMakeLists.txt

That's what I thought! What then was the original point then of putting these headers in here explicitly? Maybe it was done automatically ? Anyways, I'm sure that you thought about that, it was just confusing me.  

> I checked both the p2p and cryptonote_protocol and their files are all globed

Nice

Reply:
Without spending any time on researching it, I could imagine, that somebody first added these references here, because he was working on the daemon. Then somebody else noticed that the these individual packets don't work properly and globed them inside these packets. I've seen many things already :) Including "borrowing" entire source files, thus compiling them again, just because these people weren't feeling responsible for keeping the build files in order, or simply because they had more important things to do and "The Sprint" was reaching its end ...